ext/datastore/bigquery/adapter.go

```diff
@@ -84,6 +84,73 @@ func bqFieldModeTo(field BQField) (fieldMode, error) {
 	}
 	return fm, nil
 }
+func bqGoogleSheetsOptionsTo(m map[string]interface{}) *bqapi.GoogleSheetsOptions {
+	var skipLeadingRows int64
+	var sheetRange string
+
+	if val, ok := m["skip_leading_rows"]; ok {
+		skipLeadingRows = int64(val.(float64))
+	}
+
+	if val, ok := m["range"]; ok {
+		sheetRange = val.(string)
+	}
+
+	return &bqapi.GoogleSheetsOptions{
+		SkipLeadingRows: skipLeadingRows,
+		Range:           sheetRange,
+	}
+}
+
+func bqGoogleSheetsOptionsFrom(opt *bqapi.GoogleSheetsOptions) map[string]interface{} {
+	resultMap := make(map[string]interface{})
+
+	if opt.SkipLeadingRows != 0 {
+		// Map value of int has to be converted to float because of using interface{}
+		resultMap["skip_leading_rows"] = float64(opt.SkipLeadingRows)
```

Comment:
Seeing this for the first time, what's the problem with having `interface{}` if we don't convert it into `float64`?

Reply:
The `interface{}` type always asserts numbers from marshalled `[]byte` as `float64`. Have tried before to directly using `int64` or `int` but it seems both throws an error.
```
panic: interface conversion: interface {} is float64, not int
```
```
panic: interface conversion: interface {} is float64, not int64
```
Do you have any suggestion on this?

Reply:
Got it. Yeah JSON unmarshal will default it to float64, it's fine I guess when we unmarshal(line number 92) but when we are creating this map, we don't need to cast it to float64.